src/toxext.c

```diff
@@ -97,10 +97,13 @@ struct ToxExtExtension {
 	toxext_recv_callback recv_cb;
 	/* Callback for when we finish negotiation with a friend */
 	toxext_negotiate_connection_cb negotiate_extension_cb;
-	/* Userdata to send back when we receive a packet */
+	/* Userdata to pass back when we receive a packet */
 	void *userdata;
 };
 
+// I think this needs some consistency on terminology between "toxext packet", "packet", "toxext packet list"
```

Comment:
I think you're right. Probably in most cases I meant toxext packet list. the packet list was introduced late in the game so there's probably leftover nomenclature from the older way of thinking

Reply:
I think i was thinking of "toxext packet" being "one tox custom packet meant for toxext", in which case this is a segment of a toxext packet as well as a segment of a toxext packetlist

Reply:
Oh reading more further down, it looks like there are actually several concepts that we need to clear up the naming of.

- tox custom packet
- toxext packet buf (which is really just the data segment of a tox custom packet)
- toxext packet list (a collectino of toxext packets framed into toxext packet bufs)
- toxext packet (a single segment which is handled by a single extension)

Reply:
On further reread I think it's actually the following

- TOXEXT_SEGMENT: Segment type + segment header + extension data
- toxext packet: Sometimes same as toxext segment, sometimes same as ToxExtPacketList
- ToxExtPacketBuf: Internal representation of the toxcore custom packet. This will contain a toxext magic + 1 or many toxext segments
- ToxExtPacketList: A list of ToxExtPacketBufs. Intent here is to give a seamless abstraction to extensions for "i'd like to also send my segment". Internally we split the packet list into toxcore custom packet chunks so that when we send we can just iterate over all buffers and drop them into toxcore.

Proposed naming scheme: 
* ToxExtPacketList stays the same, however all variable names should be changed to packet_list
* ToxExtPacketBuf should be renamed to ToxCoreCustomPacketBuf to indicate that it's primary purpose is to give us a buffer to send to toxcore's custom packet API
* ToxExtPacket/segment merged into a ToxExtPacket. From an extension's perspective that means they appaned their "ToxExtPacket" onto a ToxExtPacketList"

Reply:
I like it :+1: